File: RosDom/Refactor.java
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.nio.charset.StandardCharsets;

public class Refactor {
    public static void main(String[] args) throws Exception {
        String baseDir = "c:/Programs/RosDom";
        String oldPkg = "com.example.rosdom";
        String newPkg = "ru.rosdom";
        
        File root = new File(baseDir);
        refactorDir(new File(root, "app/src/main/java"));
        refactorDir(new File(root, "app/src/test/java"));
        refactorDir(new File(root, "app/src/androidTest/java"));
        
        System.out.println("Beginning string replacements...");
        replaceStrings(root, oldPkg, newPkg);
        System.out.println("Done!");
    }
    
    static void refactorDir(File javaDir) {
        File oldPath = new File(javaDir, "com/example/rosdom");
        File newPath = new File(javaDir, "ru/rosdom");
        
        if (oldPath.exists()) {
            newPath.mkdirs();
            File[] files = oldPath.listFiles();
            if (files != null) {
                for (File f : files) {
                    System.out.println("Moving " + f.getName());
                    f.renameTo(new File(newPath, f.getName()));
                }
            }
            File current = oldPath;
            while (current.exists() && current.listFiles() != null && current.listFiles().length == 0) {
                current.delete();
                current = current.getParentFile();
            }
        }
    }
    
    static void replaceStrings(File dir, String oldStr, String newStr) throws Exception {
        File[] files = dir.listFiles();
        if (files == null) return;
        
        for (File f : files) {
            if (f.isDirectory() && !f.getName().equals("build") && !f.getName().equals(".gradle")) {
                replaceStrings(f, oldStr, newStr);
            } else if (f.isFile() && (f.getName().endsWith(".kt") || f.getName().endsWith(".kts") || f.getName().endsWith(".xml"))) {
                String content = new String(Files.readAllBytes(Paths.get(f.getAbsolutePath())), StandardCharsets.UTF_8);
                if (content.contains(oldStr)) {
                    content = content.replace(oldStr, newStr);
                    Files.write(Paths.get(f.getAbsolutePath()), content.getBytes(StandardCharsets.UTF_8));
                    System.out.println("Replaced in " + f.getAbsolutePath());
                }
            }
        }
    }
}

File: RosDom/refactor.py
import os
import shutil
import glob

base_dir = r"c:\Programs\RosDom\app\src\main\java"
test_dir = r"c:\Programs\RosDom\app\src\test\java"
androidTest_dir = r"c:\Programs\RosDom\app\src\androidTest\java"

old_pkg = "com.example.rosdom"
new_pkg = "ru.rosdom"

def refactor_dir(java_dir):
    old_path = os.path.join(java_dir, "com", "example", "rosdom")
    new_path = os.path.join(java_dir, "ru", "rosdom")
    
    if os.path.exists(old_path):
        os.makedirs(os.path.dirname(new_path), exist_ok=True)
        if not os.path.exists(new_path):
            shutil.move(old_path, new_path)
        
        # Clean up empty com/example dirs if they are empty
        try:
            os.rmdir(os.path.join(java_dir, "com", "example"))
            os.rmdir(os.path.join(java_dir, "com"))
        except:
            pass

refactor_dir(base_dir)
refactor_dir(test_dir)
refactor_dir(androidTest_dir)

# Now replace strings in all code files
replace_count = 0
for root, _, files in os.walk(r"c:\Programs\RosDom"):
    for file in files:
        if file.endswith(".kt") or file.endswith(".kts") or file.endswith(".xml"):
            fp = os.path.join(root, file)
            with open(fp, "r", encoding="utf-8") as f:
                content = f.read()
            if old_pkg in content:
                content = content.replace(old_pkg, new_pkg)
                with open(fp, "w", encoding="utf-8") as f:
                    f.write(content)
                replace_count += 1
                
print("Refactored dirs and replaced in", replace_count, "files.")
